File: smash-ranks-android/app/src/main/java/com/garpr/android/activities/TournamentsActivity.java
package com.garpr.android.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;

import com.garpr.android.App;
import com.garpr.android.R;
import com.garpr.android.misc.RegionManager;
import com.garpr.android.models.Region;

import javax.inject.Inject;

public class TournamentsActivity extends BaseActivity {

    private static final String TAG = "TournamentsActivity";

    @Inject
    RegionManager mRegionManager;


    public static Intent getLaunchIntent(final Context context) {
        return getLaunchIntent(context, null);
    }

    public static Intent getLaunchIntent(final Context context, @Nullable final Region region) {
        final Intent intent = new Intent(context, TournamentsActivity.class);

        if (region != null) {
            intent.putExtra(EXTRA_REGION, region);
        }

        return intent;
    }

    @Override
    protected String getActivityName() {
        return TAG;
    }

    @Override
    protected void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        App.get().getAppComponent().inject(this);
        setContentView(R.layout.activity_tournaments);
        setSubtitle(mRegionManager.getRegion(this).getDisplayName());
    }

    @Override
    protected boolean showUpNavigation() {
        return true;
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/activities/TournamentsActivity.kt
package com.garpr.android.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.garpr.android.R
import com.garpr.android.extensions.appComponent
import com.garpr.android.extensions.putOptionalExtra
import com.garpr.android.extensions.subtitle
import com.garpr.android.managers.RegionManager
import com.garpr.android.models.Region
import javax.inject.Inject

class TournamentsActivity : BaseActivity() {

    @Inject
    protected lateinit var regionManager: RegionManager


    companion object {
        private const val TAG = "TournamentsActivity"

        fun getLaunchIntent(context: Context, region: Region? = null): Intent {
            return Intent(context, TournamentsActivity::class.java)
                    .putOptionalExtra(EXTRA_REGION, region)
        }
    }

    override val activityName = TAG

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        appComponent.inject(this)
        setContentView(R.layout.activity_tournaments)
        subtitle = regionManager.getRegion(this).displayName
    }

}
